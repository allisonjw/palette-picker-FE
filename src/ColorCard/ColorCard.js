import React, { useState } from 'react';
import './ColorCard.scss';
import openPadlock from '../images/open-lock.png';
import closedPadlock from '../images/closed-lock.png'

const ColorCard = ({ color, locked, palette_id, lockColor}) => {

    const toggleLock = !locked ? openPadlock : closedPadlock;
    return(
        <section className="section_colorCards" style={{backgroundColor: `#${color}`}}>
              <img className="colorCard_image-lock" 
              src={toggleLock}
              alt="lock"
              onClick={() => lockColor(palette_id)} />
              <h3 className="colorCard_hex">#{color}</h3>
        </section>
    )
}

export default ColorCard;