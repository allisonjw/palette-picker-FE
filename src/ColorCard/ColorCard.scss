.section_colorCards {
    border-radius: 3px;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    margin: 1%;
    align-items: center;
}

.colorCard_image-lock {
    width: 20%;
    &:hover {
        cursor: pointer;
        transform: scale(1.1);
        transition: all .4s ease-in-out;
      }
}

.colorCard_hex {
    text-transform: uppercase;
}

