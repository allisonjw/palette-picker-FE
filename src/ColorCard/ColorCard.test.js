import React from 'react';
import { render } from '@testing-library/react';
import ColorCard from './ColorCard';
import { shallow } from 'enzyme';

describe('ColorCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ColorCard />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});