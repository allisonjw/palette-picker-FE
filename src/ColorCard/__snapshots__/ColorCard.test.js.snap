// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCard should match the snapshot 1`] = `
<section
  className="section_colorCards"
  style={
    Object {
      "backgroundColor": "#undefined",
    }
  }
>
  <img
    alt="lock"
    className="colorCard_image-lock"
    onClick={[Function]}
    src="open-lock.png"
  />
  <h3
    className="colorCard_hex"
  >
    #
  </h3>
</section>
`;
